tensorboard/notebook.py

```diff
@@ -378,18 +379,29 @@ def _display_ipython(port, height, display_handle):
       <script>
         (function() {
           const frame = document.getElementById(%JSON_ID%);
-          const url = new URL("/", window.location);
-          url.port = %PORT%;
+          const url = new URL("%ABS_PATH%", window.location);
+          %PORT_CHANGE%
```

Comment:
I think it would be more readable if we didn't use PORT_CHANGE to possibly generate an entire javascript statement.  I'd prefer if we just replaced simple types. Can your changes be refactored so that this is:

```
if ("%PORT%") {
  url.port = %PORT%;
}
```

Reply:
+1; I like this idea (though I’d suggest something like `%PORT%` being
either an int or null and setting `const port = %PORT%` rather than
using the substitution `%PORT%` once in a string context and once in a
raw context, which is a bit iffy).